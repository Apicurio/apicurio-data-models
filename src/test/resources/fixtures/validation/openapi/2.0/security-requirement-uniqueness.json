{
    "swagger":"2.0",
    "info":{
        "description":"This is a sample server Petstore server via JSON.",
        "version":"1.0.0",
        "title":"Swagger Petstore (JSON)",
        "termsOfService":"http://helloreverb.com/terms/",
        "contact":{
            "email":"apiteam@wordnik.com"
        },
        "license":{
            "name":"Apache 2.0",
            "url":"http://www.apache.org/licenses/LICENSE-2.0.html"
        }
    },
    "paths":{
        "/pet/{petId}":{
            "get":{
                "tags":[
                    "pet"
                ],
                "summary":"Find pet by ID",
                "description":"Returns a single pet",
                "operationId":"getPetById",
                "consumes":[
                    "application/x-www-form-urlencoded"
                ],
                "produces":[
                    "application/xml",
                    "application/json"
                ],
                "parameters":[
                    {
                        "name":"petId",
                        "in":"path",
                        "description":"ID of pet to return",
                        "required":true,
                        "type":"integer",
                        "format":"int64"
                    }
                ],
                "responses":{
                    "200":{
                        "description":"successful operation",
                        "schema":{
                            "$ref":"#/definitions/Pet"
                        }
                    },
                    "400":{
                        "description":"Invalid ID supplied"
                    },
                    "404":{
                        "description":"Pet not found"
                    }
                },
                "security":[
                    {
                        "api_key":[]
                    },{
                        "api_key":[]
                    },
                    {
                        "api_key":[],
                        "petstore_auth":[
                            "write:pets",
                            "read:pets"
                        ]   
                   },{
                        "petstore_auth":[
                            "read:pets",
                            "write:pets"
                       ],
                        "api_key":[]
                   },
                   {
                        "petstore_auth":[
                            "write:pets",
                            "read:pets"
                        ]
                  },
                   {
                        "petstore_auth":[
                            "write:pets",
                            "read:pets"
                        ]
                   }
               ]
            }
        }
    },
     "securityDefinitions":{
        "petstore_auth":{
            "type":"oauth2",
            "authorizationUrl":"http://petstore.swagger.io/api/oauth/dialog",
            "flow":"implicit",
            "scopes":{
                "write:pets":"modify pets in your account",
                "read:pets":"read your pets"
            }
        },
        "api_key":{
            "type":"apiKey",
            "name":"api_key",
            "in":"header"
        }
    },
    "definitions":{
        "Pet":{
            "required":[
                "name",
                "photoUrls"
            ],
            "properties":{
                "id":{
                    "type":"integer",
                    "format":"int64"
                },
                "name":{
                    "type":"string",
                    "example":"doggie"
                },
                "photoUrls":{
                    "type":"array",
                    "xml":{
                        "name":"photoUrl",
                        "wrapped":true
                    },
                    "items":{
                        "type":"string"
                    }
                },
                "status":{
                    "type":"string",
                    "description":"pet status in the store",
                    "enum":[
                        "available",
                        "pending",
                        "sold"
                    ]
                }
            },
            "xml":{
                "name":"Pet"
            }
        }
    }
}