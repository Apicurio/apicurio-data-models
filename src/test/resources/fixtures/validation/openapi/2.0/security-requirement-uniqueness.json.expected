[SREQ-005] |medium| {/paths[/pet/{petId}]/get/security[0]->security} :: Duplicate Security Requirement '[api_key]' in Security field
[SREQ-005] |medium| {/paths[/pet/{petId}]/get/security[1]->security} :: Duplicate Security Requirement '[api_key]' in Security field
[SREQ-005] |medium| {/paths[/pet/{petId}]/get/security[2]->security} :: Duplicate Security Requirement '[api_key, petstore_auth]' in Security field
[SREQ-005] |medium| {/paths[/pet/{petId}]/get/security[3]->security} :: Duplicate Security Requirement '[petstore_auth, api_key]' in Security field
[SREQ-005] |medium| {/paths[/pet/{petId}]/get/security[4]->security} :: Duplicate Security Requirement '[petstore_auth]' in Security field
[SREQ-005] |medium| {/paths[/pet/{petId}]/get/security[5]->security} :: Duplicate Security Requirement '[petstore_auth]' in Security field